### PR TITLE
Bump openPMD-api to minimal version 0.15

### PR DESCRIPTION
**Pull request by franzpoeschel**
This removes many fallback implementations

TODO:

- [x] probably need to update CI containers
       PR for CI container versions: https://codebase.helmholtz.cloud/crp/alpaka-group-container/-/merge_requests/57

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
we need to install OpenPMD by hand into the container. I will try to solve this tomorrow.

**Comment by franzpoeschel** [comment]
> we need to install OpenPMD by hand into the container. I will try to solve this tomorrow.

I have a PR for that [here](https://codebase.helmholtz.cloud/crp/alpaka-group-container/-/merge_requests/57) or is that not the right place?

File: include/picongpu/plugins/common/openPMDDefinitions.def [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#pragma once
+
+namespace picongpu::openPMD
+{
+    namespace PreferredFlushTarget
+    {
+        constexpr inline char const* const Disk = R"({"adios2": {"engine": {"preferred_flush_target": "disk"}}})";
+        constexpr inline char const* const Buffer = R"({"adios2": {"engine": {"preferred_flush_target": "buffer"}}})";
+    } // namespace PreferredFlushTarget
+} // namespace picongpu::openPMD
```
Comment (franzpoeschel) [inline include/picongpu/plugins/common/openPMDDefinitions.def]:
```suggestion
} // namespace picongpu::openPMD

```

**Comment by psychocoderHPC** [comment]
> > we need to install OpenPMD by hand into the container. I will try to solve this tomorrow.
> 
> I have a PR for that [here](https://codebase.helmholtz.cloud/crp/alpaka-group-container/-/merge_requests/57) or is that not the right place?

Thanks that's wonderful. Until we have the new containers we must compile it for any job equal to how we do it with GCC https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/share/ci/install/gcc.sh

**Comment by psychocoderHPC** [comment]
If the CI install now all correct we need to update `README.rst` and add the required openPMD-api version.

**Comment by psychocoderHPC** [comment]
> If the CI install now all correct we need to update `README.rst` and add the required openPMD-api version.

I updated the readme.

**Comment by psychocoderHPC** [comment]
@franzpoeschel The CI in this PR is now building openPMD 0.15.0. There are a few jobs crashing, could you have a look into the errors, please?

note: I opened an issue in the HZDR support system to build openPMD 0.15.2 for hemera for a modern toolchain with GCC 12.X.

**Comment by franzpoeschel** [comment]
> @franzpoeschel The CI in this PR is now building openPMD 0.15.0. There are a few jobs crashing, could you have a look into the errors, please?
> 
> note: I opened an issue in the HZDR support system to build openPMD 0.15.2 for hemera for a modern toolchain with GCC 12.X.

Thank you for your help with this
There was a little error in a file that did not get compiled in my test setup, CI is passing now

**Comment by psychocoderHPC** [comment]
I will merge this PR and update the hemera profiles in a separate PR.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**